### PR TITLE
update rust-bitcoin to 0.24.0

### PR DESCRIPTION
**Pull request by apoelstra**

### CODE REVIEW

**Comment by darosior** [comment]
Rustfmt is unhappy but otherwise lgtm

**Comment by apoelstra** [comment]
oops. ran cargo fmt

**Comment by darosior** [comment]
Hmm i think the failure on 1.22 means you need to pin CC as you did on rust-bitcoin ?
```
    # Pin `cc` for Rust 1.22
    if [ "$TRAVIS_RUST_VERSION" = "1.22.0" ]; then
        cargo generate-lockfile --verbose
        cargo update -p cc --precise "1.0.41" --verbose
    fi
```

**Comment by apoelstra** [comment]
Thanks! Cherry-picked your commit, let's see if it works now.

File: src/descriptor/create_descriptor.rs [hunk 1]
```diff
@@ -39,7 +41,7 @@ fn parse_scriptsig_top<'txin>(
     script_sig: &'txin bitcoin::Script,
 ) -> Result<(Vec<u8>, Stack<'txin>), Error> {
     let stack: Result<Vec<StackElement>, Error> = script_sig
-        .iter(true)
+        .instructions_minimal()
         .map(|instr| instr_to_stackelem(&instr))
```
Comment (dr-orlovsky) [inline src/descriptor/create_descriptor.rs:45]: Why not simply `.map(instr_to_stackelem)`? I tried locally, it compiles well with all versions, starting from MSRV

**Review by dr-orlovsky — CHANGES_REQUESTED** [review]
Propose to use this diff, which simplifies code a bit:

```diff
diff --git a/src/descriptor/create_descriptor.rs b/src/descriptor/create_descriptor.rs
index d36a7cb..6d5a08b 100644
--- a/src/descriptor/create_descriptor.rs
+++ b/src/descriptor/create_descriptor.rs
@@ -24,9 +24,9 @@ use ToPublicKey;
 /// NOTE: Miniscript pushes should only be either boolean, 1 or 0, signatures, and hash preimages.
 /// As per the current implementation, PUSH_NUM2 results in an error
 fn instr_to_stackelem<'txin>(
-    ins: &Result<Instruction<'txin>, bitcoin::blockdata::script::Error>,
+    ins: Result<Instruction<'txin>, bitcoin::blockdata::script::Error>,
 ) -> Result<StackElement<'txin>, Error> {
-    match *ins {
+    match ins {
         //Also covers the dissatisfied case as PushBytes0
         Ok(Instruction::PushBytes(v)) => Ok(StackElement::from(v)),
         Ok(Instruction::Op(opcodes::all::OP_PUSHNUM_1)) => Ok(StackElement::Satisfied),
@@ -42,7 +42,7 @@ fn parse_scriptsig_top<'txin>(
 ) -> Result<(Vec<u8>, Stack<'txin>), Error> {
     let stack: Result<Vec<StackElement>, Error> = script_sig
         .instructions_minimal()
-        .map(|instr| instr_to_stackelem(&instr))
+        .map(instr_to_stackelem)
         .collect();
     let mut stack = stack?;
     if let Some(StackElement::Push(pk_bytes)) = stack.pop() {
@@ -64,7 +64,7 @@ fn verify_p2pk<'txin>(
     if let Ok(pk) = bitcoin::PublicKey::from_slice(&pk_bytes[1..script_pubkey_len - 1]) {
         let stack: Result<Vec<StackElement>, Error> = script_sig
             .instructions_minimal()
-            .map(|instr| instr_to_stackelem(&instr))
+            .map(instr_to_stackelem)
             .collect();
         if !witness.is_empty() {
             Err(Error::NonEmptyWitness)
@@ -228,7 +228,7 @@ pub fn from_txin_with_witness_stack<'txin>(
         //bare
         let stack: Result<Vec<StackElement>, Error> = script_sig
             .instructions_minimal()
-            .map(|instr| instr_to_stackelem(&instr))
+            .map(instr_to_stackelem)
             .collect();
         if !witness.is_empty() {
             return Err(Error::NonEmptyWitness);
```

File: src/descriptor/create_descriptor.rs (hunk 1 above)
Comment (sanket1729) [reply src/descriptor/create_descriptor.rs:45]: I think this is unrelated to the intent of the PR. It is indeed a useful simplification. Can you raise another PR for this?

**Review by sanket1729 — APPROVED** [review]

**Comment by apoelstra** [comment]
Can you make a commit with these changes that I can cherry-pick?

File: src/descriptor/create_descriptor.rs (hunk 1 above)
Comment (dr-orlovsky) [reply src/descriptor/create_descriptor.rs:45]: Done: https://github.com/rust-bitcoin/rust-miniscript/pull/133

**Comment by dr-orlovsky** [comment]
@apoelstra here it is: https://github.com/rust-bitcoin/rust-miniscript/pull/133

**Pull request merged**